curated_transformers/util/fsspec.py

```diff
@@ -0,0 +1,295 @@
+    Get the configuation of a model on an fsspec filesystem.
+
+    :param fs:
+        The filesystem on which the model is stored.
+    :param model_path:
+        The path of the model on the filesystem.
+    :param fsspec_args:
+        Implementation-specific keyword arguments to pass to fsspec
+        filesystem operations.
+    :returns:
+        The model configuration.
+    """
+    config = _get_and_parse_json_file(
+        fs, model_path=model_path, filename=HF_MODEL_CONFIG, fsspec_args=fsspec_args
+    )
+    if config is None:
+        raise ValueError(
+            f"Cannot open model config path: {model_path}/{HF_MODEL_CONFIG}"
+        )
+    return config
+
+
+def get_hf_config_model_type(
+    fs: AbstractFileSystem,
+    model_path: str,
+    fsspec_args: Optional[Dict[str, Any]] = None,
+) -> str:
+    """
+    Get the type of a model on an fsspec filesystem.
+
+    :param fs:
+        The filesystem on which the model is stored.
+    :param model_path:
+        The path of the model on the filesystem.
+    :param fsspec_args:
+        Implementation-specific keyword arguments to pass to fsspec
+        filesystem operations.
+    :returns:
+        The model type.
+    """
+    config = get_model_config(fs, model_path, fsspec_args=fsspec_args)
+    model_type = config.get("model_type")
+    if model_type is None:
+        raise ValueError("Model type not found in Hugging Face model config")
+    return model_type
+
+
+def get_tokenizer_config(
+    fs: AbstractFileSystem,
+    model_path: str,
+    fsspec_args: Optional[Dict[str, Any]] = None,
+) -> Optional[Dict[str, Any]]:
+    """
+    Get the configuration of a tokenizer on an fsspec filesystem.
+
+    :param fs:
+        The filesystem on which the model is stored.
+    :param model_path:
+        The path of the model on the filesystem.
+    :param fsspec_args:
+        Implementation-specific keyword arguments to pass to fsspec
+        filesystem operations.
+    :returns:
+        Deserialized tokenizer configuration.
+    """
+    return _get_and_parse_json_file(
+        fs, model_path=model_path, filename=HF_TOKENIZER_CONFIG, fsspec_args=fsspec_args
+    )
+
+
+def get_special_tokens_map(
+    fs: AbstractFileSystem,
+    model_path: str,
+    fsspec_args: Optional[Dict[str, Any]] = None,
+) -> Optional[Dict[str, Any]]:
+    """
+    Get the special token mapping of a tokenizer on an fsspec filesystem.
+
+    :param fs:
+        The filesystem on which the model is stored.
+    :param model_path:
+        The path of the model on the filesystem.
+    :param fsspec_args:
+        Implementation-specific keyword arguments to pass to fsspec
+        filesystem operations.
+    :returns:
+        Deserialized special token_map.
+    """
+    return _get_and_parse_json_file(
+        fs, model_path=model_path, filename=SPECIAL_TOKENS_MAP, fsspec_args=fsspec_args
+    )
+
+
+def _get_and_parse_json_file(
+    fs: AbstractFileSystem,
+    *,
+    model_path: str,
```

Comment:
This looks like a generic function. Maybe we can rename this parameter to just `path`?